app/src/main/java/com/taraniuk/github/api/paging_library/ui/main/viewModel/MainActivityVIewModel.kt

```diff
@@ -0,0 +1,31 @@
+package com.taraniuk.github.api.paging_library.ui.main.viewModel
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import androidx.paging.PagingData
+import androidx.paging.rxjava2.cachedIn
+import com.taraniuk.github.api.paging_library.data.repository.InstantRepositoryImpl
+import com.taraniuk.github.api.paging_library.data.retrofit.model.Data
+import com.taraniuk.github.api.paging_library.ui.base.viewModel.BaseModel
+import javax.inject.Inject
+
+class MainActivityVIewModel @Inject constructor(private val repository: InstantRepositoryImpl) :
+    BaseModel() {
+
+    val pagingData = MutableLiveData<PagingData<Data>>()
+
+    init {
+        getResult()
+    }
+
+    private fun getResult() {
+        disposable = repository
+            .getMovies()
+            .cachedIn(viewModelScope)
+            .subscribe({
+                pagingData.value = it
+            }, {
+
```

Comment:
I think you need to handle error result to, for example just snackbar message

Reply:
Done